### DISCUSSION TITLE
Watching Repository Changes

### DISCUSSION THREAD

**Discussion by AdvaSoftware** [CodeDredd/pinia-orm#722] [Q&A] [answered]
Hi!

What's the proper practice for watching repository CRUD changes on components?

I've tried setting a watch on a computed "mapRepos" member and it seems no trigger is occuring (verified the data does change in the store).

What I current have is:

```
import XYZRepository from "..."

computed: {
    ...mapRepos({
      XYZ: XYZRepository,
    }),
  },

watch: {
    XYZ: {
      deep: true,
      handler(newValue) {}}}

```

Thank you!

**Comment by CodeDredd** [comment]
Can you please provide a [reproduction](https://v3.nuxtjs.org/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Comment by AdvaSoftware** [comment]
Thanks for the reply, I will post one later as soon as I can.

In the meanwhile, can you please refer me to how to properly watch a repository change in a vue component? I probably missed it in the docs :(

**Comment by AdvaSoftware** [comment, marked as answer]
Solved.

I've used 

`
computed: {
    XZY() {
    return store.state.value.XYZ
}
`

instead of mapRepos.
